runtime/src/tiered_storage/byte_block.rs

```diff
@@ -17,25 +24,35 @@ use {
 /// To finalize all the writes, invoke `finish` to obtain the encoded byte
 /// block.
 #[derive(Debug)]
-pub enum ByteBlockWriter {
-    Raw(Cursor<Vec<u8>>),
-    Lz4(lz4::Encoder<Vec<u8>>),
+pub struct ByteBlockWriter {
+    /// the encoder for the byte-block
+    encoder: ByteBlockEncoder,
+    /// the length of the raw data
+    len: usize,
```

Comment:
Yes, we need it because the return value of `.writer().len()` of the lz4 Encoder returns the "length of the encoded data", not the raw data, but we need the length of the raw-data.  That's why I specifically mentioned it in the comment and the PR description.

Reply:
> Ah gotcha, ok. Seems like then only the Lz4 version needs a len field. I'd probably prefer that implementation, but this way is fine and we can always change it later.

Yep, I thought about having this `usize` field under the Lz4 enum, but it's possible that we might decide to use different compressions based on the coldness (for instance, the coldest tier uses zstd, relative cold tier uses lz4).  So the current implementation offers such flexibility, but we can change it if we don't actually need more than one compression.